File: typescript/acronym/acronym.test.ts
import { parse } from "./acronym";

describe("Acronym are produced from", () => {
  it("title cased phrases", () => {
    expect(parse("Portable Network Graphics")).toEqual("PNG");
  });

  it("other title cased phrases", () => {
    expect(parse("Ruby on Rails")).toEqual("ROR");
  });

  it("inconsistently cased phrases", () => {
    expect(parse("HyperText Markup Language")).toEqual("HTML");
  });

  it("phrases with punctuation", () => {
    expect(parse("First In, First Out")).toEqual("FIFO");
  });

  it("other phrases with punctuation", () => {
    expect(parse("PHP: Hypertext Preprocessor")).toEqual("PHP");
  });

  it("phrases with punctuation and sentence casing", () => {
    expect(parse("Complementary metal-oxide semiconductor")).toEqual("CMOS");
  });
});

File: scala/acronym/src/test/scala/AcronymTest.scala
import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers


/** @version 1.6.0 */
class AcronymTest extends AnyFunSuite with Matchers {

  test("basic") {
    Acronym.abbreviate("Portable Network Graphics") should be ("PNG")
  }

  test("lowercase words") {
    Acronym.abbreviate("Ruby on Rails") should be ("ROR")
  }

  test("punctuation") {
    Acronym.abbreviate("First In, First Out") should be ("FIFO")
  }

  test("all caps word") {
    Acronym.abbreviate("GNU Image Manipulation Program") should be ("GIMP")
  }

  test("punctuation without whitespace") {
    Acronym.abbreviate("Complementary metal-oxide semiconductor") should be ("CMOS")
  }

  test("very long abbreviation") {
    Acronym.abbreviate("Rolling On The Floor Laughing So Hard That My Dogs Came Over And Licked Me") should be ("ROTFLSHTMDCOALM")
  }

  test("consecutive delimiters") {
    Acronym.abbreviate("Something - I made up from thin air") should be ("SIMUFTA")
  }

  test("apostrophes") {
    Acronym.abbreviate("Halley's Comet") should be ("HC")
  }
}
